backend/engine/plugins/gitsecrets/secrets_processor.py

```diff
@@ -2,56 +2,150 @@
 import os
 import re
 from string import Template
+from typing import Optional
 
+from engine.plugins.gitsecrets.secret_regex import SecretRegex
 from engine.plugins.lib.utils import setup_logging
 
 log = setup_logging("gitsecrets")
 
 DEFAULT_SEPARATOR = ":"
 DEFAULT_SETTER_ERROR_MESSAGE = "property $property is read only"
 
+URL_REGEX = r"://[^/\s:@]{3,64}:[^/\s:@]{3,64}@[a-zA-Z0-9\.-]+(:[0-9]{1,5}){0,1})"
+
 # !!! IMPORTANT
 # This regex needs to match the git-secrets regexes in Dockerfiles/data/secret-patterns
 REGEXES = [
-    re.compile(str(r"(amzn\.mws\.[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")),
-    re.compile(str(r"(AIza[0-9A-Za-z_-]{35})")),
-    re.compile(str(r"(https://hooks.slack.com/services/T[a-zA-Z0-9_]{8}/B[a-zA-Z0-9_]{8}/[a-zA-Z0-9_]{24})")),
-    re.compile(str(r"(AKIA[A-Z0-9]{16})")),
-    re.compile(str(r"(ASIA[A-Z0-9]{16})")),
-    re.compile(str(r"(ACCA[A-Z0-9]{16})")),
-    re.compile(str(r"(EAACEdEose0cBA[0-9A-Za-z]{16,255})")),
-    re.compile(str(r"([f|F][a|A][c|C][e|E][b|B][o|O][o|O][k|K](.){1,64}['|\"][0-9a-f]{32}['|\"])")),
-    re.compile(str(r"([a|A][p|P][i|I][_]?[k|K][e|E][y|Y](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
-    re.compile(str(r"([s|S][e|E][c|C][r|R][e|E][t|T](.){1,64}['|\"][0-9a-zA-Z]{32,45}['|\"])")),
-    re.compile(str(r"([0-9]+-[0-9A-Za-z_]{32}\.apps\.googleusercontent\.com)")),
-    re.compile(str(r"(ya29\.[0-9A-Za-z\-_]{55})")),
-    re.compile(
-        str(
-            (
-                r"((ftp|ftps|http|https|mongodb|mongodb\+srv|postgres)"
-                r"://[^/\s:@]{3,64}:[^/\s:@]{3,64}@[a-zA-Z0-9\.-]+(:[0-9]{1,5}){0,1})"
+    SecretRegex(
```

Comment:
Pulls the types from https://github.com/dxa4481/truffleHogRegexes/blob/9257f45e66e31ad3fefd9d6e2b2e09dfb83043d6/truffleHogRegexes/regexes.json

Which I believe is where we originally got these regular expressions